src/theory/arith/linear/arith_static_learner.cpp

```diff
@@ -338,9 +346,201 @@ void ArithStaticLearner::addLearnedLemma(TNode n,
 
 std::shared_ptr<ProofNode> ArithStaticLearner::getProofFor(Node fact)
 {
-  // proofs not yet supported
+  Trace("arith-static-pf") << "Prove: " << fact << std::endl;
+  std::vector<Node> antec;
+  Node conc = fact;
+  // maps arithmetic variables to their constraints in the antecedant
+  std::map<Node, Node> amap;
+  if (fact.getKind() == Kind::IMPLIES)
+  {
+    if (fact[0].getKind() == Kind::AND)
+    {
+      antec.insert(antec.end(), fact[0].begin(), fact[0].end());
+    }
+    else
+    {
+      antec.push_back(fact[0]);
+    }
+    conc = fact[1];
+    for (const Node& a : antec)
+    {
+      Assert(a.getNumChildren() == 2 && a[1].isConst());
+      amap[a[0]] = a;
+    }
+  }
+  Kind ck = conc.getKind();
+  Assert(conc.getNumChildren() == 2 && conc[0].getKind() == Kind::ITE);
+  NodeManager* nm = nodeManager();
   CDProof cdp(d_env);
-  cdp.addTrustedStep(fact, TrustId::ARITH_STATIC_LEARN, {}, {});
+  Node cond = conc[0][0];
+  Node truen = nm->mkConst(true);
+  if (cond.getNumChildren() == 2 && cond[0] == conc[0][2]
```

Comment:
document the different shapes that `cond` and `conc` can have and how the if-then-else treats them